Add V0 type uint8_t (able to flag standard V0s vs cascade V0s)

ddobrigk opened the pull request.

ddobrigk committed:
Add V0 bool if standalone V0 cuts passed

alibuild committed:
Please consider the following formatting changes

ddobrigk committed:
Merge pull request #26 from alibuild/alibot-cleanup-12435

Please consider the following formatting changes to #12435

ddobrigk committed:
Change to bitmask

alibuild committed:
Please consider the following formatting changes

ddobrigk committed:
Merge pull request #27 from alibuild/alibot-cleanup-12435

Please consider the following formatting changes to #12435

ddobrigk commented:
@shahor02 I changed this such that there is a uint8_t bitmask in the V0 data model. The point is that like this we will also be able to tag special cases such as the photon hypothesis exceptions from @f3sch, which could potentially make everything quite safe (and unlocks the ability to safely do more exotic things if necessary). The actual checks can be dynamic columns defined on this bitmask to make stuff more intuitive.

shahor02 commented:
@ddobrigk OK, I've added new getter `isPhotonOnly()` : currently it is set when V0 is using TPC-only tracks and therefore tested only against photon mass hypothesis.

ddobrigk committed:
Add isConversionCandidate dynamic

ddobrigk committed:
Update AnalysisDataModel.h

Changed to IsPhotonV0 to keep consistency with other getter (following discussion with Felix)

alibuild commented:
Error while checking build/O2/fullCI for 8e7f832a8a2e333e828e2cfb6c1aae437489b949 at 2023-12-11 20:38:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
task timeout reached .. killing all processes


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/aafd33c9e4c1f0c6b470f3259e3e23c3e343fee1/slc8_x86-64/o2checkcode/1.0-local270/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12435/8e7f832a8a2e333e828e2cfb6c1aae437489b949/build_O2_fullCI/pretty.html).

shahor02 commented:
Failures unrelated
